Borodin Ilya

ilyaborodin opened the pull request:
Прошу оставить какие-либо комментарии по оформлению кода. До этого работал там, где не важна была чистота кода и оформление, а важен был только результат. Поэтому приму замечания с удовольствием)

Последние сутки? Пора браться за задания)

Сделано:

- 1-ая итерация
- 2-ая итерация
- 3-ая итерация
- 4-ая итерация
- 5-ая итерация
- 6-ая итерация(выполнена частично, описание в readme)
- Задания по практическим занятиям
- Задания по гиту

ilyaborodin committed:
Делаем комит с пустым файлом

ilyaborodin committed:
Add 1st iter

ilyaborodin committed:
Изменение пути на интепретатор

ilyaborodin committed:
Выполнил 2-ую итерацию

Добавлен файл setup.py
Добавлен метож main
Изменена версия программы на "2.0"

ilyaborodin committed:
Переменовал команду в rss-reader

AlexeiBuzuma commented on App/Args_parser.py:
```diff
@@ -0,0 +1,29 @@
+
+
+def parsing_args():
+    """Парсим аргументы"""
```
комментарии на русском лучше, чем транслитом, но на английском языке они лучше всего :)

ilyaborodin replied:
Принял)

AlexeiBuzuma commented on App/Portal.py:
```diff
@@ -0,0 +1,68 @@
+        except Exception as e:
+            print("Problems with rss processing")
+            logging.error(str(e))
+            exit()
```
у программы такого размера может быть одна точка выхода, в данном случае их минимум 7 
тут имеет смысл сделать одну точку выхода из програмы (примерно рядом с точкой входа)

ilyaborodin replied:
Поправил, спасибо

ilyaborodin committed:
Убрал лишние точки выхода и поправил систему ошибок

Убрал все лишние exit
Добавил прокидывание ошибок дальше по вертикали

ilyaborodin committed:
Добавил назввание портала к каждой новости

ilyaborodin committed:
Добавил класс saver

ilyaborodin committed:
Убрал limit с print и добавил его в метод update

ilyaborodin committed:
Выполнена 3-ая итерация

ilyaborodin committed:
создал Readme

ilyaborodin committed:
Update README.md

ilyaborodin committed:
Создал папку для хранения данных

ilyaborodin committed:
Выполнена 4-ая итерация

Добавлены классы ToPDF и ToHtml
Добавлены новые аругемнты для конвертации
Изменена версия программы

ilyaborodin committed:
Update README.md

ilyaborodin committed:
Убрал лишний импорт

ilyaborodin committed:
Merge remote-tracking branch 'origin/Final_Task' into Final_Task

ilyaborodin committed:
Перевел документацию на английский

ilyaborodin committed:
Выполнил 5-ую итерацию

ilyaborodin committed:
Update README.md

ilyaborodin committed:
Update README.md

ilyaborodin committed:
Добавил \n к выводу новостей

ilyaborodin committed:
Merge remote-tracking branch 'origin/Final_Task' into Final_Task

AlexeiBuzuma commented on App/News.py:
```diff
@@ -0,0 +1,75 @@
+        self.links = []
+        self.clear_text()
+
+    def pars_date(self, struct):
```
А чем `datetime` или аналогичные библиотеки не подходит для этой задачи? 
или я что-то не до конца понимаю

AlexeiBuzuma commented on App/News.py:
```diff
@@ -0,0 +1,75 @@
+            logging.warning("Problems with tag parsing:\n" + str(e))
+
+    def __str__(self):
+        string = "Channel name: {0}\n" \
```
Это мое субъективное мнение, но мне кажется, что это слишком "тяжелая" логика для метода "__str__" 
Лично я бы вынес это в отдельный метод и не использовал для этих задач __str__ 
но это мое мнение :)

AlexeiBuzuma commented on App/RSSListener.py:
```diff
@@ -0,0 +1,48 @@
+
+
+class RSSListener:
+    """"""
```
док стринг

AlexeiBuzuma commented on App/Saver.py:
```diff
@@ -0,0 +1,54 @@
+                            delete_list.append(new_d)
+                for new_d in delete_list:
+                    date_handler[date].remove(new_d)
+                with open("./Cache/" + date, 'wb') as f:
```
в различных операционных системах могут отличаться разделители в пути. Советую не хардкодить такие вещи, а пользоваться функционалом `os.path`

ilyaborodin committed:
Выполнил частично 6-ую итерацию

ilyaborodin committed:
Update README.md

AlexeiBuzuma commented:
```bash
# rss-reader https://news.tut.by/rss/world.rss --help
unsupported format character 'y' (0x79) at index 49
The program suddenly completed its work
```
